src/main/java/org/springframework/data/convert/PropertyValueConverterFactory.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.data.convert;
+
+import java.util.Arrays;
+import java.util.List;
+
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.data.convert.PropertyValueConverterFactories.BeanFactoryAwarePropertyValueConverterFactory;
+import org.springframework.data.convert.PropertyValueConverterFactories.CachingPropertyValueConverterFactory;
+import org.springframework.data.convert.PropertyValueConverterFactories.ChainedPropertyValueConverterFactory;
+import org.springframework.data.convert.PropertyValueConverterFactories.ConfiguredInstanceServingValueConverterFactory;
+import org.springframework.data.convert.PropertyValueConverterFactories.SimplePropertyConverterFactory;
+import org.springframework.data.mapping.PersistentProperty;
+import org.springframework.lang.Nullable;
+import org.springframework.util.Assert;
+
+/**
+ * A factory that provides {@link PropertyValueConverter value converters}.
+ * <p>
+ * Depending on the applications need {@link PropertyValueConverterFactory factories} can be {@link #chained(List)
+ * chained} and the resulting {@link PropertyValueConverter converter} may be
+ * {@link #caching(PropertyValueConverterFactory) cached}.
+ *
+ * @author Christoph Strobl
+ * @since 2.7
+ */
+public interface PropertyValueConverterFactory {
+
+	/**
+	 * Get the {@link PropertyValueConverter} applicable for the given {@link PersistentProperty}.
+	 *
+	 * @param property must not be {@literal null}.
+	 * @param <A> domain specific type.
+	 * @param <B> store native type.
+	 * @param <C> value conversion context to use.
+	 * @return can be {@literal null}.
+	 */
+	@Nullable
+	@SuppressWarnings("unchecked")
+	default <A, B, C extends ValueConversionContext<?>> PropertyValueConverter<A, B, C> getConverter(
+			PersistentProperty<?> property) {
+
+		if (!property.hasValueConverter()) {
+			return null;
+		}
+
+		return getConverter((Class<PropertyValueConverter<A, B, C>>) property.getValueConverterType());
+	}
+
+	/**
+	 * Get the converter by its type.
+	 *
+	 * @param converterType must not be {@literal null}.
+	 * @param <A> domain specific type.
+	 * @param <B> store native type.
+	 * @param <C> value conversion context to use.
+	 * @return
+	 */
+	@Nullable
+	<A, B, C extends ValueConversionContext<?>> PropertyValueConverter<A, B, C> getConverter(
```

Comment:
Let's make the `getPropertyValueConverter()` method non-nullable and introduce a `hasConverter` method to check for the presence of converters.